curvefs/src/metaserver/s3compact_manager.h

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * @Project: curve
+ * @Date: 2021-09-09
+ * @Author: majie1
+ */
+
+#ifndef CURVEFS_SRC_METASERVER_S3COMPACT_MANAGER_H_
+#define CURVEFS_SRC_METASERVER_S3COMPACT_MANAGER_H_
+
+#include <memory>
+#include <vector>
+
+#include "curvefs/src/copyset/copyset_node_manager.h"
+#include "curvefs/src/metaserver/s3compact.h"
+#include "src/common/configuration.h"
+#include "src/common/s3_adapter.h"
+
+using curve::common::Configuration;
+using curve::common::RWLock;
+using curve::common::S3Adapter;
+
+namespace curvefs {
+namespace metaserver {
+
+class S3CompactWorkQueueImpl;
+class S3CompactManager {
+ private:
+    std::shared_ptr<S3Adapter> s3Adapter_;
+    std::shared_ptr<Configuration> conf_;
+    std::shared_ptr<S3CompactWorkQueueImpl> s3compactworkqueueImpl_;
+    std::shared_ptr<S3Compact> s3compacts_;
```

Comment:
wrong type ? should it be `std::vector<S3Compact>` ?

Reply:
fixed.